source/kraken/realtime.cpp

```diff
@@ -528,31 +564,21 @@ void handle_realtime(const std::string& id,
     auto log = log4cplus::Logger::getInstance("realtime");
     LOG4CPLUS_TRACE(log, "realtime trip update received");
 
-    if (! is_handleable(trip_update)
+    if (! is_handleable(trip_update, *data.pt_data, is_realtime_add_enabled, is_realtime_add_trip_enabled)
             || ! check_trip_update(trip_update)) {
         LOG4CPLUS_DEBUG(log, "unhandled real time message");
         return;
     }
-    // if is_realtime_add_enabled = false, this disables the adding trip option too
-    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
-        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
-        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
-        return;
-    }
-    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
-        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
-                << trip_update.trip().trip_id()
-                << " with ADDITIONAL_SERVICE effect");
-        return;
-    }
 
-    auto meta_vj = data.pt_data->meta_vjs.get_mut(trip_update.trip().trip_id());
-    if (! meta_vj) {
-        LOG4CPLUS_ERROR(log, "unknown vehicle journey " << trip_update.trip().trip_id());
-        // TODO for trip().ADDED, we'll need to create a new VJ
-        return;
+    // Get or create meta VJ
+    navitia::type::MetaVehicleJourney* meta_vj = nullptr;
+    if (! data.pt_data->meta_vjs.exists(trip_update.trip().trip_id())) {
+        LOG4CPLUS_DEBUG(log, "unknown vehicle journey, create Meta VJ " << trip_update.trip().trip_id());
+        meta_vj = data.pt_data->meta_vjs.emplace(trip_update.trip().trip_id());
+        meta_vj->tz_handler = data.pt_data->tz_manager.get_first_timezone();
```

Comment:
Let's add a TODO and list it in what should be properly managed.

Reply:
To create a Meta VJ you need to add a new in _pt_data->meta_vjs_ (with _emplace_ method) and create a new _meta_vj->tz_handler_.
What do you want to doc and where ?

Reply:
If I understand correctly, the TZ you chose is maybe not the correct one, we have to pick the right one (don't know how we have to choose it: main one from the coverage?).
So I'd add a `TODO: pick a meaningful TZ` in code, and keep track of it in a JIRA ticket with everything we should choose more carefully than "additional service" for example :wink:

Reply:
okay I'll add a TODO comment for that